### PR TITLE
chore(deps): add 'syncpack' and sync dependencies

### PR DESCRIPTION
**Pull request by danroc**
This PR adds the [syncpack](https://jamiemason.github.io/syncpack/) tool to our workflow to ensure that our dependencies are kept in sync inside the monorepo.

Replaces #52

### CODE REVIEW

**Comment by danroc** [comment]
@SocketSecurity ignore npm/ws@7.4.6

False positive, there is a resolution for it: https://github.com/MetaMask/accounts/blob/8d88b6797f93db7cf5d1ebb407c49caf356dee09/package.json#L45

**Comment by danroc** [comment]
@SocketSecurity ignore npm/axios@1.7.3

False positive, there is a resolution for it: https://github.com/MetaMask/accounts/blob/8d88b6797f93db7cf5d1ebb407c49caf356dee09/package.json#L44

**Comment by danroc** [comment]
@SocketSecurity ignore npm/supports-hyperlinks@2.3.0

Both authors are still listed on the projects page on [NPM](https://www.npmjs.com/package/supports-hyperlinks?activeTab=versions).

**Comment by danroc** [comment]
@SocketSecurity ignore npm/bech32@1.1.4

After checking the project's page on [NPM](https://www.npmjs.com/package/bech32/v/1.1.4?activeTab=versions) and [GitHub](https://github.com/bitcoinjs/bech32), both authors seem to contribute to the project.

**Comment by danroc** [comment]
@SocketSecurity ignore npm/@ethersproject/providers@5.7.2

It's an expected capability for a provider.

File: packages/keyring-snap-bridge/package.json [hunk 1]
```diff
@@ -32,26 +32,26 @@
...
+    "jest-it-up": "^3.1.0",
+    "rimraf": "^5.0.7",
+    "ts-jest": "^29.0.5",
+    "ts-node": "^10.9.2",
+    "typedoc": "^0.25.13",
     "typescript": "~4.8.4"
   },
   "peerDependencies": {
-    "@metamask/keyring-api": "^8.1.1"
+    "@metamask/keyring-api": "8.1.2"
```
Comment (ccharly) [inline packages/keyring-snap-bridge/package.json:54]: Should this be using caret notation?

```suggestion
    "@metamask/keyring-api": "^8.1.2"
```

**Review by ccharly — APPROVED** [review]

**Pull request merged**